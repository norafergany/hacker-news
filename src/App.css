.light {
    background-color: #E5E5E5;
}

.nav {
    align-items: center;
    display: inline-flex;
    gap: 1rem;
    margin-top: 40px;
}

/*//TODO add top space under nav*/

.content {
    padding-left: 8em;
}

.nav-heading {
    font-weight: bold;
    /*font-size:2em;*/
    color: #000;

}

.fa-hacker-news {
    /*background:#fff;*/
    color: coral;
}

html > {
    color: #000;
}

body {
    font-family: 'Verdana', 'Geneva',
    sans-serif;

    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
}


/* Initial CSS

 */
/*.App {*/
/*  text-align: center;*/
/*}*/

/*.App-logo {*/
/*  height: 40vmin;*/
/*  pointer-events: none;*/
/*}*/

/*@media (prefers-reduced-motion: no-preference) {*/
/*  .App-logo {*/
/*    animation: App-logo-spin infinite 20s linear;*/
/*  }*/
/*}*/

/*.App-header {*/
/*  background-color: #282c34;*/
/*  min-height: 100vh;*/
/*  display: flex;*/
/*  flex-direction: column;*/
/*  align-items: center;*/
/*  justify-content: center;*/
/*  font-size: calc(10px + 2vmin);*/
/*  color: white;*/
/*}*/

/*.App-link {*/
/*  color: #61dafb;*/
/*}*/

/*@keyframes App-logo-spin {*/
/*  from {*/
/*    transform: rotate(0deg);*/
/*  }*/
/*  to {*/
/*    transform: rotate(360deg);*/
/*  }*/
/*}*/
