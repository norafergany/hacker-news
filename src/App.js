import React, {useEffect, useState} from "react";
import {ErrorBoundary} from "react-error-boundary";
import {ThemeProvider} from "styled-components";
import {GlobalStyles} from "./components/GlobalStyles";
import {darkTheme, lightTheme} from "./components/Themes";
import "./App.css";
import NavBar from "./components/NavBar";
import ArticleListContainer from "./components/ArticleListContainer";

export const App = () => {

    const [theme, setTheme] = useState('light');



    const toggleTheme = (theme) => {
        const newTheme = theme === 'light' ? 'dark' : 'light';
        setTheme(newTheme);
        localStorage.setItem('theme',newTheme);

    }

    const errorFallback = ({error, resetErrorBoundary}) => {
            return (
                <div role="alert">
                    <p>Something went wrong:</p>
                    <pre>{error.message}</pre>
                    <button onClick={resetErrorBoundary}>Try again</button>
                </div>
            )
//remember last setting on refresh
    }

    useEffect(() => {
        function getInitialTheme() {
            let storedTheme = 'light';

            if (localStorage.getItem('theme')) {
                storedTheme = localStorage.getItem('theme');
            }
            else if((window.matchMedia("(prefers-color-scheme: dark)").matches)) {
                storedTheme = 'dark';
            }
            const initialTheme = storedTheme;

            // const storedTheme = localStorage.getItem('theme') || (window.matchMedia("(prefers-color-scheme: dark)").matches ? "dark" : "light");
            setTheme(initialTheme);
            localStorage.setItem('theme', initialTheme);
        }
        getInitialTheme();
        // const initialTheme = storedTheme ? storedTheme : theme;
        // localStorage.setItem('theme', initialTheme);

    }, [])

        return (
            // article lsit container to ber wrapped by error boundary so that it can inherit methods from errorboundary for fetching errors
            <>
                <ThemeProvider theme={theme === 'light' ? lightTheme : darkTheme} handleClick={toggleTheme}>
                    <>
                        <GlobalStyles/>
                {/*<div  theme={this.state.theme} className={`container`} >*/}
                <div className={`container`}>
                    {/*<NavBar />*/}

                    <NavBar theme={theme} handleClick={() =>toggleTheme(theme)}/>
                        <div className="content" >
                            <ErrorBoundary FallbackComponent={errorFallback}>
                                <ArticleListContainer />

                            </ErrorBoundary>

                        </div>
                </div>
                        </>
                </ThemeProvider>
            </>

        );
}

export default App;

