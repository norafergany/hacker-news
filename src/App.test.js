import React from "react";
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import { createRoot } from 'react-dom/client';



import App from './App';
import MatchMediaMock from "jest-matchmedia-mock";

let matchMedia;
/**
 * @jest-environment jsdom
 */
describe('App module', () => {
  beforeAll(() => {
    matchMedia = new MatchMediaMock();
  })

  it('renders without crashing', () => {
    render(<App />);
  });

})


it('matches the snapshot', () => {
  const component = renderer.create(<App />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})