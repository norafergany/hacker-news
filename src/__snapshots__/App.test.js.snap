// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="container"
>
  <header>
    <nav
      className="my-3 navbar navbar-expand-lg navbar-light"
    >
      <div
        className="container-fluid"
      >
        <a
          className="navbar-brand"
          href="."
        >
          <i
            className="fa-brands fa-hacker-news fa-2x logo me-2"
          />
          <span
            className="h1 link-dark"
          >
            Hacker News
          </span>
        </a>
        <button
          className="sc-bdVaJa liGjSw theme-button ms-auto me-1"
          onClick={[Function]}
        >
          <i
            aria-label="light-display-icon"
            className="fa-solid fa-moon"
          />
        </button>
      </div>
    </nav>
  </header>
</div>
`;
